mutable_tree.go

```diff
@@ -263,6 +263,17 @@ func (tree *MutableTree) LoadVersion(targetVersion int64) (int64, error) {
 	return latestVersion, nil
 }
 
+// LoadVersionOverwrite returns the version number of the latest version found.
+// Higher versions' data will be deleted.
+func (tree *MutableTree) LoadVersionForOverwriting(targetVersion int64) (int64, error) {
+	latestVersion, err := tree.LoadVersion(targetVersion)
+	if err != nil {
+		return latestVersion, err
+	}
+	tree.deleteVersionsFrom(targetVersion+1)
+	return targetVersion, nil
```

Comment:
According to the comment it "returns the version number of the latest version found". Here we just return the argument passed by the caller `targetVersion`?

Reply:
Sorry. It is my mistake. I will change the comment

Reply:
No worries! Thank you. But then: why do we even need to return this value if it is alreadz known bz the caller. It is not even read in the tests. It's a bit confusing.

Reply:
If the targetVersion is greater than the lastest version, then the lastest version will be return. I have modified the test the check the return value.

Reply:
Thanks! It's still a bit confusing. So it actually returns the latest version together with an error. How is this useful? Usually, in go if you return an error the other return values aren't considered by the caller (usually nil or default values). But I see that this is what `LoadVersion` already handles it like this. I think it would be useful if it returned a particular which the caller could check against. But this is probably orthogonal to this PR. No need to change this in this PR.